sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mlme.c

```diff
@@ -40,31 +40,47 @@ static void _build_join_req_pkt(uint8_t *appeui, uint8_t *deveui,
     lorawan_hdr_set_mtype((lorawan_hdr_t *)hdr, MTYPE_JOIN_REQUEST);
     lorawan_hdr_set_maj((lorawan_hdr_t *)hdr, MAJOR_LRWAN_R1);
 
-    le_uint64_t l_appeui = *((le_uint64_t *)appeui);
+    le_uint64_t l_joineui = *((le_uint64_t *)joineui);
     le_uint64_t l_deveui = *((le_uint64_t *)deveui);
 
-    hdr->app_eui = l_appeui;
+    hdr->join_eui = l_joineui;
     hdr->dev_eui = l_deveui;
 
     le_uint16_t l_dev_nonce = *((le_uint16_t *)dev_nonce);
 
     hdr->dev_nonce = l_dev_nonce;
 
-    gnrc_lorawan_calculate_join_mic(psdu, JOIN_REQUEST_SIZE - MIC_SIZE, appkey,
-                                    &hdr->mic);
+    gnrc_lorawan_calculate_join_req_mic(psdu, JOIN_REQUEST_SIZE - MIC_SIZE, key,
+                                        &hdr->mic);
 }
 
 void gnrc_lorawan_trigger_join(gnrc_lorawan_t *mac)
 {
-    iolist_t pkt = {.iol_base = mac->mcps.mhdr_mic, .iol_len =
-                    sizeof(lorawan_join_request_t), .iol_next = NULL};
-    gnrc_lorawan_send_pkt(mac, &pkt, mac->last_dr);
+    iolist_t pkt = { .iol_base = mac->mcps.mhdr_mic, .iol_len =
+                         sizeof(lorawan_join_request_t), .iol_next = NULL };
+
+    gnrc_lorawan_send_pkt(mac, &pkt, mac->last_dr,
+                          gnrc_lorawan_pick_channel(mac));
 }
 
 static int gnrc_lorawan_send_join_request(gnrc_lorawan_t *mac, uint8_t *deveui,
-                                          uint8_t *appeui, uint8_t *appkey,
-                                          uint8_t dr)
+                                          uint8_t *eui, uint8_t *key, uint8_t dr)
 {
+
+#if IS_ACTIVE(CONFIG_GNRC_LORAWAN_1_1)
+    /**
+    * DevNonce starting at 0 when device is powered up and incremented with every
+    * Join-request.
+    *
+    * TODO: A DevNonce value SHALL NEVER be reused for a given JoinEUI value.
+    * If the end-device can be power-cycled then DevNonce SHALL be persistent
+    * (stored in a non-volatile memory)
+    */
+    ((le_uint16_t *)mac->mlme.dev_nonce)->u16++;
```

Comment:
since we are using `le_xxx` structures to make the code portable, we should probably also keep these operations portable. Therefore, please use `byteorder_xxx` functions here instead of just operating on the u16 member.